### PR TITLE
GHA to publish VSIX package

### PR DESCRIPTION
**Pull request by khawkins**
I haven't been able to test the final (publishing) step, for obvious reasons. But I've tested the rest, by substituting the final step with a step that unzips the downloaded package and lists its contents, ensuring that we are in fact downloading a valid package from our release.

### CODE REVIEW

**Comment by khawkins** [comment]
At a high level, this manual workflow goes like this:

1. User goes to the Actions tab, selects the "Publish Extension Package for GitHub Release" action.
2. They click the "Run workflow" button, which will ask for "Release Tag Name / Version". Our release tags are formatted as `vx.y.z`, but the input will allow for either that _or_ just the version (i.e. `x.y.z`).
3. The script will validate the tag format (errors out if it's invalid), then attempt to interrogate a Release associated with that tag name. Errors out if that Release does not exist.
4. It will then grab the well-known VSIX package name asset associated with that release. Errors out if that asset does not exist.
5. Finally, it will publish the VSIX package.

File: .github/workflows/publish_release.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                      }
+
+                      // Officially supported SemVer version syntax regex.
+                      // See https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+                      const validVersionRegex = /^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/;
+                      if (!validVersionRegex.test(tagNameValue)) {
+                          throw new Error(`'${tagNameValue}' is not a valid version string.`);
+                      }
+
+                      // Put the 'v' back on, as our release tags are formatted as 'v'x.y.z.
```
Comment (dbreese) [inline .github/workflows/publish_release.yml:42]: Nice comments!

**Review by dbreese — APPROVED** [review]
LGTM!

**Review by sfdctaka — APPROVED** [review]
LGTM!

File: .github/workflows/publish_release.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                        throw new Error(`Failed to download '${vsixPackageName}': ${response.status}`);
+                      }
+                      const fs = require('fs');
+                      fs.writeFileSync(vsixPackageName, Buffer.from(response.data, 'binary'));
+
+            - name: Publish to VSCode Marketplace
+              env:
+                  VSCE_PERSONAL_ACCESS_TOKEN: ${{ secrets.VSCE_PERSONAL_ACCESS_TOKEN }}
+              run: |
+                  npx vsce publish -p ${{ env.VSCE_PERSONAL_ACCESS_TOKEN }} --packagePath ${{ steps.get_package_name.outputs.result }}
```
Comment (gbockus-sf) [inline .github/workflows/publish_release.yml:110]: heh... a lot of this GHA flow is new to me...so LGTM!

**Pull request merged**